### PR TITLE
3node-mini-provisioning: Post-Install, Prepare, Config, Init phases modifications

### PR DESCRIPTION
**Pull request by udayan-y21**
# Backend

## Problem Statement
<pre>
Story Ref (if any):
</pre>
## Unit testing on RPM done
<pre>
Yes/No
</pre>
## Problem Description
<pre>

</pre>
## Solution
<pre>

</pre>
## Unit Test Cases
<pre>

</pre>
Signed-off-by:

### CODE REVIEW

File: cicd/auxiliary/csm_cicd.sh [hunk 1]
```diff
@@ -28,17 +28,33 @@ yum install -y $RPM_PATH/*.rpm
...
+pip3 freeze
+cat /etc/machine-id
+MID=$(cat /etc/machine-id)
+echo $MID
 chmod 777 -R /var/log/seagate
-csm_setup post_install --config json:///opt/seagate/cortx/csm/templates/csm_setup_conf_template.json
-csm_setup config --config json:///opt/seagate/cortx/csm/templates/csm_setup_conf_template.json
-csm_setup init --config json:///opt/seagate/cortx/csm/templates/csm_setup_conf_template.json
+
+sed -i -e 's|TMPL_MACHINE_ID|$(cat /etc/machine-id)|g' "/opt/seagate/cortx/csm/conf/csm.post_install.tmpl.1-node"
```
Comment (pawankumarsrivastava) [inline cicd/auxiliary/csm_cicd.sh:43]: Are we not considering 3 node template here ?

File: csm/common/network_addresses.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        Get virtual host IP address.
+
+        :return: String representation of virtual host IP address
+        """
+        try:
+            ip = str(Conf.get(const.CSM_GLOBAL_INDEX, 'PROVISIONER>virtual_host'))
+            ip_address(ip)
+            return ip
+        except ValueError as e:
+            desc = 'Could not obtain management IP address from CSM config'
```
Comment (pawankumarsrivastava) [inline csm/common/network_addresses.py:40]: Could not obtain management IP address from CSM config -> Unable to obtain Virtual host from the configuration.
Please have the description generic.

File: csm/common/service_urls.py [hunk 3]
```diff
@@ -27,58 +28,53 @@ class ServiceUrls:
...
+    @staticmethod
+    def get_mgmt_url() -> str:
         """
         Return the management URL for the appliance.
 
         :returns: management URL as a string.
         """
-        network_configuration = await self._provisioner.get_network_configuration()
         ssl_check = Conf.get(const.CSM_GLOBAL_INDEX, 'CSM_SERVICE>CSM_WEB>ssl_check')
+        scheme = 'https' if ssl_check else 'http'
```
Comment (pawankumarsrivastava) [inline csm/common/service_urls.py:39]: use const class for hardcoded strings

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: cicd/auxiliary/csm_cicd.sh [hunk 4]
```diff
@@ -28,17 +28,33 @@ yum install -y $RPM_PATH/*.rpm
 mkdir -p /etc/ssl/stx/ /etc/cortx/ha/
 cp -f $CSM_REPO_PATH/cicd/auxiliary/stx.pem /etc/ssl/stx/
 #cp -f $CSM_REPO_PATH/cicd/auxiliary/etc/database.json /etc/cortx/ha/
-
+yum install -y http://cortx-storage.colo.seagate.com/releases/eos/uploads/rhel/rhel-7.7.1908/csm_uploads/cortx-py-utils-1.0.0-177_8cc273a.noarch.rpm
```
[the remark this answers is not in the document]
Comment (udayan-y21) [reply cicd/auxiliary/csm_cicd.sh:31]: yes I will remove it. Its a part of testing

File: cicd/auxiliary/csm_cicd.sh (hunk 1 above)
Comment (udayan-y21) [reply cicd/auxiliary/csm_cicd.sh:43]: This will be removed as it was part of local testing

**Comment by udayan-y21** [comment]
Closing this PR deu to unable to resolve conflicts. Created new PR https://github.com/Seagate/cortx-manager/pull/420/files

**Pull request closed without merging**